src/swaps/hop/HopSwapProvider.ts

```diff
@@ -0,0 +1,579 @@
+        asset.code === tokenName ||
+        asset.matchingAsset === tokenName
+      ) {
+        return tokenName
+      }
+    } else {
+      // erc20 asset
+      if (
+        tokens[chain]?.l1CanonicalToken?.toLowerCase() === asset?.contractAddress.toLowerCase() ||
+        tokens[chain]?.l2CanonicalToken?.toLowerCase() === asset?.contractAddress.toLowerCase()
+      ) {
+        return tokenName
+      }
+    }
+  }
+
+  _getSendInfo(assetFrom: any, assetTo: any, hop: any) {
+    if (!assetFrom || !assetTo) return null
+    const _chainFrom = this.getChain(assetFrom.chain)
+    const _chainTo = this.getChain(assetTo.chain)
+    if (!_chainFrom || !_chainTo) return null
+    const availableToken = this._getAllTokens(hop)
+    let _from, _to
+    for (const token in availableToken) {
+      if (!_from) _from = this._findAsset(assetFrom, _chainFrom.slug, availableToken[token], token)
+      if (!_to) _to = this._findAsset(assetTo, _chainTo.slug, availableToken[token], token)
+    }
+    if (!_from || !_to || _from !== _to) return null
+    const supportedAssetsFrom = hop.getSupportedAssetsForChain(_chainFrom.slug)
+    const supportedAssetsTo = hop.getSupportedAssetsForChain(_chainTo.slug)
+    if (!supportedAssetsFrom[_from] || !supportedAssetsTo[_to]) return null
+    return { bridgeAsset: _from, chainFrom: _chainFrom, chainTo: _chainTo }
+  }
+
+  // eslint-disable-next-line no-unused-vars
+  public async getQuote({ network, from, to, amount } : QuoteRequest) {
+    if (amount <= new BN(0)) return null
+    const assetFrom = cryptoassets[from]
+    const assetTo = cryptoassets[to]
+    const fromAmountInUnit = currencyToUnit(cryptoassets[from], new BN(amount))
+    const hop = this._getHop(network)
+    if (!hop || !hop.isValidChain(assetFrom.chain) || !hop.isValidChain(assetTo.chain)) return null
+    const info = this._getSendInfo(assetFrom, assetTo, hop)
+    if (!info?.bridgeAsset || !info?.chainFrom || !info?.chainTo) return null
+    const { bridgeAsset, chainFrom, chainTo } = info
+    const bridge = hop.bridge(bridgeAsset)
+    const sendData = await bridge.getSendData(fromAmountInUnit.toString(), chainFrom, chainTo)
+    if (!sendData) return null
+    return {
+      from,
+      to,
+      // Amounts should be in BigNumber to prevent loss of precision
+      fromAmount: fromAmountInUnit.toFixed(),
+      toAmount: new BN(sendData.amountOut.toString()).toFixed(),
+      hopAsset: bridgeAsset,
+      hopChainFrom: chainFrom,
+      hopChainTo: chainTo,
+      receiveFee: new BN(sendData.adjustedBonderFee.toString())
+        .plus(new BN(sendData.adjustedDestinationTxFee.toString()))
+        .toString()
+    }
+  }
+
+  async _approveToken(bridge: any, chainFrom: any, fromAmount: any, signer: any, fee: any) {
+    const txData = await bridge.populateSendApprovalTx(fromAmount, chainFrom)
+    const approveTx = await signer.sendTransaction({
+      ...txData,
+      gasPrice:
```

Comment:
only gas price? What if it's EIP1559? Not sure how other swap providers work in this regard

Reply:
All the other amms like uniswap, oneinch, sovryn are working the same way - only gas price